bluenews_rednews/newsfeed/lib/newsfeed_web/commanders/page_commander.ex

```diff
@@ -0,0 +1,142 @@
+defmodule NewsfeedWeb.PageCommander do
+  use Drab.Commander
```

Comment:
I think you're the first project I've seen using Drab. Very cool.

You've made me think about how this kind of code should be structured. It seems to me that the commander is basically a controller, so I'd ideally like it to be short and focussed only on intermediating between the browser and the business components. 

I was looking at, for instance, 

~~~ elixir
  defhandler url_searchbtn_clicked(socket, _sender) do
    set_style(socket, "#other-sources", display: "block")
     reset(socket)
     # reset other input
    set_prop(socket, "#kwd-user-input", value: "")
     query = query(socket, "#url-user-input", :value)
    case query do
      {:ok, values} ->
        user_input = values["#url-user-input"]["value"]
        article = ArticleProcessing.process_article(user_input)
         keywords = article.keywords
         put_store(socket, :keywords, keywords)
        articles = SimilarArticleRetrieval.get_all_articles(keywords)
         set_default_articles(articles, socket)
       _ ->
        nil
    end
  end
~~~
I got to thinking that there might be an argument for a separate View module that handled UI specific stuff (such as resetting fields after a button press). I also thought that adding a `article_keywords` function to `ArticleProcessing` would tidy things a little:

~~~ elixir
defhandler url_searchbtn_clicked(socket, _sender) do
  View.reset_after_search_button(socket)
  query(socket, "#url-user-input", :value)
  |> case do
    {:ok, values} ->
      article_url = values["#url-user-input"]["value"]
      keywords = ArticleProcessing.article_keywords(article_url)
      put_store(socket, :keywords, keywords)
      SimilarArticleRetrieval.get_all_articles(keywords)
      |> set_default_articles(socket)
     _ ->
      nil
  end
end
~~~

But this is all uncharted territory. I'm mentioning it mostly to attempt to influence future coding decisions you might make. The maxim is "divide and conquer", but I also feel it could be "divide and clarify."